prime-router/src/testIntegration/kotlin/transport/RESTTransportIntegrationTests.kt

```diff
@@ -373,4 +380,22 @@ class RESTTransportIntegrationTests : TransportIntegrationTests() {
         val retryItems = mockRestTransport.send(transportType, header, reportId, null, context, actionHistory)
         assertThat(retryItems).isNotNull()
     }
+
+    @Test
+    fun `test flexion rest transport`() {
+        val header = makeHeader()
+        val mockRestTransport = spyk(RESTTransport(mockClientPostOk()))
+        every { mockRestTransport.lookupDefaultCredential(any()) }.returns(
+            UserApiKeyCredential("flexion", "123")
+        )
+        every { runBlocking { mockRestTransport.getAuthTokenWithUserApiKey(any(), any(), any(), any()) } }.returns(
```

Comment:
Is there a way we can test getAuthTokenWithUserApiKey without mocking it? RESTTransport.kt doesn't have any code coverage, because most of it is mocked. Maybe we can create an additional ticket to add that. But, I'm concerned that we don't have enough testing to make sure this doesn't break other receivers that use REST. Is there a way we can test this doesn't break that before merging?

Reply:
@victor-chaparro, thank you for reviewing the PR.  To verify the Transport working with ETOR/Trusted-Intermediary, I run the Trusted-Intermediary and ReportStream server locally with the setting provided in 0149-etor.yml (uncommented out the localhost).  Then, I manually send patient resource to the reportStream's localhost /api/report endpoint and check for a successful response.  However, this is not what I really want for testing the ETOR use case.  I agreed with you that the getAuthTokenWithUserApiKey needs to be tested without mock.  I will create a ticket to do so. 

For code coverage, I use IntelliJ run the test with the Coverage command and check for path coverage.

Reply:
Can we also test that it works for other receivers that use REST locally?

Reply:
Yes, I can and will test with NY and OK to make certain that it works with them as well.

Reply:
Tested with NY and it work well.
![image](https://github.com/CDCgov/prime-reportstream/assets/20068283/91d38943-9ec6-49c9-a685-c6f13b71d988)

Reply:
Test cases to cover getAuthTokenWithUserApiKey are added to RESTTransportIntegrationTest.kt.